std/container/slist.d

```diff
@@ -123,6 +123,19 @@ struct SList(T)
         return n;
     }
 
+    private static Node * findNodeByValue(Node * n, T value)
+    {
+        assert(n);
+        auto ahead = n._next;
+        while (ahead && ahead._payload != value)
+        {
+            n = ahead;
+            enforce(n);
```

Comment:
Why do you test `n` / `ahead` here? The loop body is only executed if `ahead` is evaluated to `true` (i.e. not `null`).